### PR TITLE
Fix inventory messages when nereid_catalog_variants is present #7571

### PR DESCRIPTION
**Pull request by PritishC**

### CODE REVIEW

**Comment by PritishC** [comment]
screen_010

File: templates/webshop/product.jinja [hunk 1]
```diff
@@ -404,13 +407,38 @@
           {'variant': product}
         ));
 
+        // Product inventory
+        if(product.inventory_status[0] == 'in_stock'){
+          statusClass = iconClass ='info';
+          // Remove the disabled attribute from the buy now button.
+          $('#buy-now-btn').removeAttr('disabled');
+        } else {
+          statusClass = 'danger';
```
Comment (tarunbhardwaj) [inline templates/webshop/product.jinja]: Add `disabled` Attr to buy button

Comment (PritishC) [inline templates/webshop/product.jinja]: Done.

File: templates/webshop/product.jinja [hunk 2]
```diff
@@ -404,13 +409,39 @@
           {'variant': product}
         ));
 
+        // Product inventory
+        if(product.inventory_status[0] == 'in_stock'){
+          statusClass = iconClass ='info';
```
Comment (tarunbhardwaj) [inline templates/webshop/product.jinja]: I think this `class` logic can be moved to _js template only

**Comment by tarunbhardwaj** [comment]
LGTM

File: product.py [hunk 3]
```diff
@@ -29,6 +29,22 @@ class Product:
...
+        Downstream implementation which adds a key called inventory_status
+        to the dictionary if purpose is 'variant_selection'.
+
+        :param purpose: String which decides structure of dictionary
+        """
+        res = super(Product, self).serialize(purpose=purpose)
+
+        if purpose == 'variant_selection':
+            res.update({
+                'inventory_status': self.inventory_status(),
```
Comment (tarunbhardwaj) [inline product.py]: if this method is in nereid-catalog-variant then this implementation should be there only.

Comment (PritishC) [inline product.py]: nereid-catalog-variants is not dependent on nereid-cart-b2c. Hence the `serialize` method had to be overriden here itself.

**Pull request merged**